processor/src/main/java/org/mapstruct/ap/internal/model/NestedPropertyMappingMethod.java

```diff
@@ -52,17 +54,27 @@ public Builder mappingContext(MappingBuilderContext mappingContext) {
 
         public NestedPropertyMappingMethod build() {
             List<String> existingVariableNames = new ArrayList<>();
+            Parameter sourceParameter = null;
             for ( Parameter parameter : method.getSourceParameters() ) {
                 existingVariableNames.add( parameter.getName() );
+                if ( sourceParameter == null && !parameter.isMappingTarget() && !parameter.isMappingContext() ) {
+                    sourceParameter = parameter;
+                }
             }
             final List<Type> thrownTypes = new ArrayList<>();
             List<SafePropertyEntry> safePropertyEntries = new ArrayList<>();
+            if ( sourceParameter == null ) {
+                throw new IllegalStateException( "Method " + method + " has no source parameter." );
+            }
+
+            String previousPropertyName = sourceParameter.getName();
```

Comment:
is it `previousPropertyName` or `originaSourceParameterName`

Reply:
It is actually both. The idea is to replace

```
<#macro localVarName index><#if index == 0>${sourceParameter.name}<#else>${propertyEntries[index-1].name}</#if></#macro>